File: Recursion/ReverseString/ReverseString.scala
object Solution {
    def reverseString(s: Array[Char]): Unit = {
        
        def loop(first: Int, second: Int): Unit = {
            if (first > second) ()
            else {
                val tmp = s(first)
                s(first) = s(second)
                s(second) = tmp
                loop(first + 1, second - 1)
            } 
        
        };
        loop(0, s.length - 1)
    }
}

File: Recursion/ReverseString/ReverseString.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: List[str]
        :rtype: None Do not return anything, modify s in-place instead.
        """
        def helper(first, second):
            if first > second:
                return
            s[first], s[second] = s[second], s[first]
            helper(first + 1, second - 1)
        
        helper(0, len(s) -1 )